port/unix/omrsysinfo.c

```diff
@@ -3968,6 +4022,67 @@ getHandleOfCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subs
 	return rc;
 }
 
+/**
+ * Reads the Cgroup metric (if has multiple searched for metric key and reads it) and returns the value to the param `value`
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary
+ * @param[in] subsystemFlag flag of type OMR_CGROUP_SUBSYSTEMS_* representing the cgroup subsystem
+ * @param[in] fileName name of the file under cgroup subsystem
+ * @param[in] metricKeyInFile name of the cgroup metric which is in a group of metrics in a cgroup subsystem file
+ * @param[in/out] value pointer to char * which stores the value for the specified cgroup subsystem metric
+ *
+ * @return 0 on success, negative error code on any error
+ */
+static int32_t
+readCgroupMetricFromFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFlag, const char *fileName, const char *metricKeyInFile, char *value, char **fileContent, BOOLEAN isSingleLineKeyValue)
+{
+	int32_t rc = 0;
+	FILE *file = NULL;
+	rc = getHandleOfCgroupSubsystemFile(portLibrary, subsystemFlag, fileName, &file);
+	if (0 != rc) {
+		goto _end;
+	}
+	if (NULL == metricKeyInFile) { /* If `metricKeyInFile` is NULL then it states the file has only one Cgroup metric to be read */
+		if (NULL == fgets(value, MAX_LINE_LENGTH, file)) {
+			rc = OMRPORT_ERROR_FILE_OPFAILED;
+		}
+		goto _end;
+	} else if (isSingleLineKeyValue) { /* Else if its a single lined entry with metric, we search for the metric and get its value */
+		char bufferStore[MAX_LINE_LENGTH] = {0};
+		size_t size = strlen(metricKeyInFile);
+		if (NULL != fgets(bufferStore, MAX_LINE_LENGTH, file)) {
+			char *tmpPtr = bufferStore;
+			if (0 == strncmp(tmpPtr, metricKeyInFile, size)) {
+				tmpPtr += size;
+				sscanf(tmpPtr, "%s", value);
+				goto _end;
+			}
+		} else {
+			rc = OMRPORT_ERROR_FILE_OPFAILED;
+			goto _end;
+		}
+	} else { /* Else we read the file to a buffer and pick the metrics from it later */
+		*fileContent = portLibrary->mem_allocate_memory(portLibrary, 1024, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+		if (NULL != *fileContent) {
+			size_t returnvalue = fread(*fileContent, 1024, 1, file);
+			if (0 != returnvalue) {
```

Comment:
Thanks for pointing this bug @DanHeidinga . I have made changes to the code accordingly. As the `returnvalue` depends on the total elements read and it can't be always `1024`, I'm checking it against `0` (Case where nothing is read) `0 == returnValue`